src/main/java/rx/internal/operators/OperatorDistinctUntilChanged.java

```diff
@@ -62,7 +62,7 @@ public OperatorDistinctUntilChanged(Func2<? super U, ? super U, Boolean> compara
     
     @Override
     public Boolean call(U t1, U t2) {
-        return (t1 == t2 || (t1 != null && t1.equals(t2)));
+        return (t1 == t2 || (t2 != null && t2.equals(t1)));
```

Comment:
The current comparison is completely legal, same as Java 7's Objects.equals(). Why did you swap the variables? Implementation of equals should be symmetric: a.equals(b) == b.equals(a).